webapp/packages/core-connections/src/DatabaseAuthDialog/DBAuthDialogFooter.tsx

```diff
@@ -9,41 +9,68 @@
 import { observer } from 'mobx-react-lite';
 import styled, { css } from 'reshadow';
 
-import { Button } from '@cloudbeaver/core-blocks';
+import { Button, ErrorMessage } from '@cloudbeaver/core-blocks';
 import { useTranslate } from '@cloudbeaver/core-localization';
+import type { GQLErrorCatcher } from '@cloudbeaver/core-sdk';
+import { composes, useStyles } from '@cloudbeaver/core-theming';
 
-const styles = css`
-  controls {
-    display: flex;
-    height: 100%;
-    flex: 1;
-    align-items: center;
-    margin: auto;
-    justify-content: flex-end;
-  }
-`;
+const styles = composes(
+  css`
+    ErrorMessage {
+      composes: theme-background-secondary from global;
+    }
+  `,
+  css`
+    controls {
+      display: flex;
+      height: 100%;
+      flex: 1;
+      align-items: center;
+      justify-content: flex-end;
+    }
+    ErrorMessage {
+      flex: 0.9;
+      & + controls {
+        margin-left: 17px;
+        flex: 0.1;
+      }
+    }
+`);
 
 export interface Props {
   isAuthenticating: boolean;
   onLogin: () => void;
+  error?: GQLErrorCatcher;
+  onShowDetals?: () => void;
 }
 
 export const DBAuthDialogFooter = observer(function DBAuthDialogFooter({
   isAuthenticating,
   onLogin,
+  error,
+  onShowDetals,
 }: Props) {
   const translate = useTranslate();
 
-  return styled(styles)(
-    <controls as="div">
-      <Button
-        type="button"
-        mod={['unelevated']}
-        disabled={isAuthenticating}
-        onClick={onLogin}
-      >
-        {translate('authentication_login')}
-      </Button>
-    </controls>
+  return styled(useStyles(styles))(
+    <>
+      {error?.responseMessage && (
+        <ErrorMessage
+          text={error.responseMessage}
+          hasDetails={error.hasDetails}
+          onShowDetails={onShowDetals}
+        />
+      )}
+      <controls>
+        <Button
+          type="button"
+          mod={['unelevated']}
+          disabled={isAuthenticating}
+          onClick={onLogin}
+        >
+          {translate('authentication_login')}
+        </Button>
+      </controls>
+    </>
   );
```

Comment:
seems like, will be better to pass `ErrorMessage` as children to `DBAuthDialogFooter`